### PR TITLE
Blob adapter

### PR DESCRIPTION
**Pull request by Elvis-Shi**
Blob adapter document(toc not added yet).

### CODE REVIEW

File: biztalk/core/azure-blob-adapter.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    | **Blob container name** | Select the name of your Blob container from the drop-down list which should be automatically manipulated after the **Connection string** field is filled up. |
+    | **Blob name** | Name of the blob used by adapter to upload the message to Blob storage container. Macros can be used for Blob name as well as file adapter, for available Macros please refere to [Using macros in file names](restrictions-when-configuring-the-file-adapter.md#using-macros-in-file-names). |
+    | **Namespace for blob metadata** | Context properties of the message can be written to blob metadata if namespace of the property match this field. |
+
+    When finished, your properties look similar to the following: 
+
+    ![General properties](../core/media/BlobAdapter-send-General.png)
+
+5. Configure the **Advanced** properties:
+    |Use this|To do this|
```
Comment (msnehagup) [inline biztalk/core/azure-blob-adapter.md:68]: this table is not showing up properly in the Preview

Comment (msnehagup) [reply biztalk/core/azure-blob-adapter.md:68]: https://github.com/msnehagup/biztalk-docs/pull/5

File: biztalk/core/azure-blob-adapter.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    | **Resource Group** | Select your resource group that has your Azure storage account |
+
+4. Configure the **General** properties: 
+
+    |Use this|To do this|  
+    |---|---|  
+    | **Storage Authentication** | Select an authentication method. <ul><li>Typically, it's recommended to use a Shared Access Signature, which is also by default selected. You can input the Shared Access Signature connection string to **Connection string** field to provide the authentication.</li> <li>If you are using Access keys as authentication method, a collection of storage account will be poped up in **Account** drop-down list, after you select the storage account, **Connection string** field will be automatically filled up using your primary access key, which also know as **key1**. </li></ul><br />The  The following links are good resources to help you decide which is right for your scenario:<br/><br/>[About Azure storage accounts](https://docs.microsoft.com/azure/storage/common/storage-create-storage-account)<br/>[Using shared access signatures (SAS)](https://docs.microsoft.com/azure/storage/common/storage-dotnet-shared-access-signature-part-1) |
```
Comment (msnehagup) [inline biztalk/core/azure-blob-adapter.md]: The linked page for 'About Azure storage accounts' doesn't seem very relevant here. It doesn't talk about anything related to authentication

File: biztalk/core/azure-blob-adapter.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    | **Resource Group** | Select your resource group that has your Azure storage account |
+
+4. Configure the **General** properties: 
+
+    |Use this|To do this|  
+    |---|---|  
+    | **Storage Authentication** | Select an authentication method. <ul><li>Typically, it's recommended to use a Shared Access Signature, which is also selected by default. You can input the Shared Access Signature connection string in the **Connection string** field to provide the authentication.</li> <li>If you are using Access keys as authentication method, a collection of storage account will be populated up in **Account** drop-down list. After you select the storage account, **Connection string** field will be automatically populated with your primary access key, which also known as **key1**. </li></ul><br />The following links are good resources to help you decide which authentication method is right for your scenario:<br/><br/>[About Azure storage accounts](https://docs.microsoft.com/azure/storage/common/storage-create-storage-account)<br/>[Using shared access signatures (SAS)](https://docs.microsoft.com/azure/storage/common/storage-dotnet-shared-access-signature-part-1) |
```
Comment (SanjivGupta-SanGupt) [inline biztalk/core/azure-blob-adapter.md]: This is somewhat confusing without a background in Storage.

File: biztalk/core/azure-blob-adapter.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    | **Blob container name** | Select the name of your Blob container from the drop-down list. The list should be auto-populated after the **Connection string** field is filled up. |
+    | **Blob name** | Name of the blob used by adapter to upload the message to Blob storage container. Macros can be used for Blob name similar to file adapter. For available Macros please refer to [Using macros in file names](restrictions-when-configuring-the-file-adapter.md#using-macros-in-file-names). |
```
Comment (SanjivGupta-SanGupt) [inline biztalk/core/azure-blob-adapter.md]: If we must mention file adapter, lets make its case as in the documentation for this adapter and provide a hyper link. Question is, if we are providing a reference to Using macros in file names, do we need to mention file adapter?

File: biztalk/core/azure-blob-adapter.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    When finished, your properties look similar to the following: 
+
+    ![General properties](../core/media/BlobAdapter-send-General.png)
+
+5. Configure the **Advanced** properties:
+
+    |Use this|To do this|  
+    |---|---|  
+    |**Blob type**| Azure Blob adapter can upload message as different blob types as needed, visit [blob types](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction) for more information. |
+    |**Write mode**| In case of same blob name already exists, different **Write mode** are provided. |
```
Comment (SanjivGupta-SanGupt) [inline biztalk/core/azure-blob-adapter.md]: 1) Even if it is trivial, we must mention the different write modes
2) Possible to improve English articulation. How about - Use different **Write mode** settings to handle cases when given **Blob name** already exists

File: biztalk/core/azure-blob-adapter.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+    ```xml
+    <Data>
+      <DataID>DataID_0</DataID>
+      <DataDetails>DataDetails_0</DataDetails>
+    </Data>
+    ```
+
+4. Start the File receive location and the Azure Blob storage send port.
+5. Copy **AzureBlobStorageMessage.xml** sample message into the receive folder (C:\Temp\In\). The send port sends the XML file to the Azure Blob storage. You can verify that by looking into your Azure storage container and exam the new created or updated file.
```
Comment (SanjivGupta-SanGupt) [inline biztalk/core/azure-blob-adapter.md]: examin

File: biztalk/core/azure-blob-adapter.md [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+4. Configure the **Azure Account** properties: 
+
+    |Use this|To do this|  
+    |---|---|  
+    | **Sign-in** | Sign into your Azure account |
+    | **Subscription** | Select the subscription that has your Azure storage account |
+    | **Resource Group** | Select your resource group that has your Azure storage account |
+
+4. Configure the **Endpoint** properties: 
+
```
Comment (SanjivGupta-SanGupt) [inline biztalk/core/azure-blob-adapter.md:115]: Something does not seem right here related to Even hub?

File: biztalk/core/azure-blob-adapter.md (hunk 4 above)
Comment (Elvis-Shi) [reply biztalk/core/azure-blob-adapter.md]: Using macros in file names is not a standalone page, it is a section of "Restrictions when configuring the File adapter" page, so i feel it is more natual to just say this is similiar to file adapter .

File: biztalk/core/azure-blob-adapter.md (hunk 2 above)
Comment (Elvis-Shi) [reply biztalk/core/azure-blob-adapter.md]: replace it with another more helpful link.

File: biztalk/core/azure-blob-adapter.md (hunk 7 above)
Comment (Elvis-Shi) [reply biztalk/core/azure-blob-adapter.md:115]: Good catch. Seems i messed up here when we were exploring the correct way to work with github. Corrected.

File: biztalk/core/azure-blob-adapter.md (hunk 3 above)
Comment (Elvis-Shi) [reply biztalk/core/azure-blob-adapter.md]: I understand, this is a common scenario, everyone document actually have some Prerequisites for reader. The whole article structure i followed event hub adapter, unless you are proposing somehting else.

File: biztalk/core/azure-blob-adapter.md (hunk 5 above)
Comment (SanjivGupta-SanGupt) [reply biztalk/core/azure-blob-adapter.md]: No, I did not mean language improvements. 
What are the values of the Write Mode? And when should the reader use a specific value.

**Review by SanjivGupta-SanGupt — APPROVED** [review]

**Pull request merged**